pkg/cloud/ibmcloud/deployer.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ibmcloud
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	clustercommon "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/cluster-api/pkg/util"
+)
+
+const ProviderName = "ibmcloud"
+const (
+	IBMCloudIPAnnotationKey = "ibmcloud-ip-address"
+	IBMCloudIdAnnotationKey = "ibmcloud-resourceId"
+)
+
+func init() {
+	clustercommon.RegisterClusterProvisioner(ProviderName, NewDeploymentClient())
+}
+
+type DeploymentClient struct{}
+
+func NewDeploymentClient() *DeploymentClient {
+	return &DeploymentClient{}
+}
+
+func (d *DeploymentClient) GetIP(cluster *clusterv1.Cluster, machine *clusterv1.Machine) (string, error) {
```

Comment:
@xunpan can you please add some comments here, like 

```
// TODO: Need to consolidate this with https://github.com/multicloudlab/cluster-api-provider-ibmcloud/blob/master/pkg/cloud/ibmcloud/actuators/machine/actuator.go#L161-L208
```

@jichenjc comments?

Reply:
sorry, i had a talk with @xunpan just now, and saw this after merge action
let's keep discussing and add additional PR if needed